### PR TITLE
Introduce makefile to run build/release commands

### PR DESCRIPTION
**Pull request by cpanato**
- Introduce makefile to run build/release commands like the build/test and release, also inject the ldflags and fix the version command

```
$ ./apko version
     _      ____    _  __   ___
    / \    |  _ \  | |/ /  / _ \
   / _ \   | |_) | | ' /  | | | |
  / ___ \  |  __/  | . \  | |_| |
 /_/   \_\ |_|     |_|\_\  \___/
apko

GitVersion:    v0.1
GitCommit:     98e1549
GitTreeState:  dirty
BuildDate:     '2022-02-24T16:08:23Z'
GoVersion:     go1.17.6
Compiler:      gc
Platform:      darwin/amd64
```

on the image
```
$ docker run ko.local/apko:feca0f036129cdc1a62bb2a0a3f1b0852e826d075a57c4b5a7120031b0d455d7 version
     _      ____    _  __   ___
    / \    |  _ \  | |/ /  / _ \
   / _ \   | |_) | | ' /  | | | |
  / ___ \  |  __/  | . \  | |_| |
 /_/   \_\ |_|     |_|\_\  \___/
apko

GitVersion:    v0.1
GitCommit:     98e1549
GitTreeState:  dirty
BuildDate:     '2022-02-24T16:08:23Z'
GoVersion:     go1.17.6
Compiler:      gc
Platform:      linux/amd64
```

cc @mattmoor @kaniini

### CODE REVIEW

File: mageFile.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	gitVersion := getVersion()
+	gitCommit := getCommit()
+	os.Setenv("LDFLAGS", generateLDFlags())
+	os.Setenv("COSIGN_EXPERIMENTAL", "true")
+
+	if os.Getenv("KO_DOCKER_REPO") == "" {
+		return errors.New("missing KO_DOCKER_REPO environment variable")
+	}
+
+	digest, err := sh.Output("ko", "build", "--base-import-paths", "--bare",
```
Comment (mattmoor) [inline mageFile.go:66]: --base-... and --bare are mutually exclusive

File: mageFile.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	return sh.RunV("ko", "apply", "-BRf", "config/")
+}
+
+// BuildImagesLocal build images locally and not push
+func BuildImagesLocal() error {
+	gitVersion := getVersion()
+	gitCommit := getCommit()
+	os.Setenv("LDFLAGS", generateLDFlags())
+
+	return sh.RunV("ko", "publish", "--base-import-paths", "--bare", "--local",
```
Comment (mattmoor) [inline mageFile.go:50]: We're mixing publich here and build below

File: mageFile.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+		"--platform=linux/amd64", "--tags", gitVersion, "--tags", gitCommit, "--tags", "latest",
```
Comment (mattmoor) [inline mageFile.go:51]: For local, we may want to key off of GOARCH?

File: mageFile.go (hunk 2 above)
Comment (mattmoor) [reply mageFile.go:50]: also same comment about base/bare

File: mageFile.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+	ldFlags := generateLDFlags()
+
+	fmt.Println(ldFlags)
+	return sh.RunV("go", "build", "-trimpath", "-ldflags", ldFlags, "-o", "apko", "./cmd/apko/main.go")
+}
+
+func KoApply() error {
+	os.Setenv("LDFLAGS", generateLDFlags())
+
+	return sh.RunV("ko", "apply", "-BRf", "config/")
```
Comment (mattmoor) [inline mageFile.go:41]: nit: we expand out `-B` (`--base-import-paths`) et al below.

File: mageFile.go (hunk 1 above)
Comment (cpanato) [reply mageFile.go:66]: humm did not know and they not complain, which one we should use?

**Review by mattmoor — COMMENTED** [review]
IDK how folks feel about the mageFile stuff in general.  It's certainly new to me.

A few general comments

File: mageFile.go (hunk 1 above)
Comment (mattmoor) [reply mageFile.go:66]: I generally use the "base" one, but I used `--bare` for the release build so `KO_DOCKER_REPO` is exactly where the image goes.

Comment (cpanato) [reply mageFile.go:66]: ok, thanks

File: mageFile.go (hunk 3 above)
Comment (cpanato) [reply mageFile.go:51]: sgtm

File: mageFile.go (hunk 4 above)
Comment (cpanato) [reply mageFile.go]: expanded to make it clear

File: mageFile.go (hunk 2 above)
Comment (cpanato) [reply mageFile.go]: change to use build

**Comment by cpanato** [comment]
> IDK how folks feel about the mageFile stuff in general. It's certainly new to me.
> 
> A few general comments

if people do not like I can change to a traditional makefile

**Comment by cpanato** [comment]
@kaniini thanks for the feedback i will update that

**Comment by cpanato** [comment]
changed to makefile PTAL @kaniini @mattmoor

**Comment by kaniini** [comment]
Looks good!

**Pull request merged**